### PR TITLE
Elena torgasheva week1

### PR DESCRIPTION
**Pull request by Elena17T**

### CODE REVIEW

File: week1/project/ecomerce/src/App.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+import {useState} from 'react';
+import Category from './components/Category';
+import Products from './components/Products';
+
+function App() {
+  const [category, setCategory] = useState('All');
+  return (
+    <div className="App">
+      <h1>Products</h1>
+      <Category useStateCategory={setCategory} />
```
Comment (ObadaMoustafa) [inline week1/project/ecomerce/src/App.js]: naming is very important .. `useStateCategory` is for custom hooks which you will study later how to create a hook yourself so using `use` keyword will confuse you or your team when you read it.

✏️need to modify✏️

File: week1/project/ecomerce/src/components/Category.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    const getCategoryName =  name.slice(6);
+    const checkCategoryState = (previousState) => {
+      if (getCategoryName === previousState) {
+        return "All";
+      } else {
+        return getCategoryName;
+      }
+    };
+    props.useStateCategory(checkCategoryState);
+  }
```
Comment (ObadaMoustafa) [inline week1/project/ecomerce/src/components/Category.js]: smart 🚀💥

File: week1/project/ecomerce/src/components/Category.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  
+  function ListCategory({name, onClick}) {
+    return(
+        <button className="categoty-btn" onClick={onClick}>{name}</button>
+    );
```
Comment (ObadaMoustafa) [inline week1/project/ecomerce/src/components/Category.js]: it's much better to create this component in another file

**Review by ObadaMoustafa — COMMENTED** [review]
great work @Elena17T there is only one thing you need to change but totally it's great

File: week1/project/ecomerce/src/App.js (hunk 1 above)
Comment (Elena17T) [reply week1/project/ecomerce/src/App.js]: changed

File: week1/project/ecomerce/src/components/Category.js (hunk 3 above)
Comment (Elena17T) [reply week1/project/ecomerce/src/components/Category.js]: created new page with button functional component

**Pull request closed without merging**